src/main/java/filter/SemanticException.java

```diff
@@ -0,0 +1,11 @@
+package filter;
+
+import filter.expression.QualifierType;
+
+@SuppressWarnings("serial")
+public class SemanticException extends FilterException {
+    public SemanticException(QualifierType type) {
+        super(String.format("Invalid input in filter for type \"%s\"", type));
```

Comment:
Ok will work making the preamble shorter. The explanation can be universal like the input is not supported because a more specific explanation like what type of inputs are only supported would be too long to fit in the status bar also